### DISCUSSION TITLE
Ask your questions here

### DISCUSSION THREAD

**Discussion by Ruben2776** [Ruben2776/PicView#35] [Q&A]
If you have any questions about PicView, support, feature requests, feedback etc., you can ask them here.

**Comment by tawoirgard** [comment]
Hello, 
I love this app, its simplicity and its design. Well done !!!
The only thing I would like to ask : I tried to find a way to clear history automatically when exiting the program but in vain. No way to even do this manually.
Maybe the feature exists but I did not see where to toggle it... If not, is this feature in your todo list ? 

Best regards,   Thierry-Antoine

**Reply by Ruben2776** [reply]
Thanks for your kind words.

I can look into implementing clearing history in a future update.

**Comment by thomasbkdk** [comment]
Does this program have 'always on top' and is it possible to insert a picture from windows clip board?

**Reply by Ruben2776** [reply]
Press `T` to toggle staying on top or click the cog icon in the lower toolbar and toggle the option for `Stay on top of other windows`

Pressing `Ctrl + V` inserts from your clipboard which supports images, URL's, files and file paths

**Comment by FrederikVerscheure** [comment]
Hi
I love this tool and i use it on multiple computers but on one computer there's a bug in viewing images.
The image is shown but only partially and the rest is blurred transparant
any suggestions
![image](https://github.com/Ruben2776/PicView/assets/59872689/208d46c9-fef0-4397-ae18-152a47decb46)

**Reply by Ruben2776** [reply]
If possible, can you send me a copy of the image and I will investigate it

**Comment by FrederikVerscheure** [comment]
Hi Ruben,

The problem is not the image file, since the picture opens on another computer in PicView without any problems.
Any idea what I can check? I use the portable version.

Grt
Fred

**Comment by FrederikVerscheure** [comment]
Hi,
Apologies for the long wait, bit I was out due to illness.
Attached is a screenshot of what the users sees when he opens a picture from Teamcenter (our PLM platform running on java).
Picview is displayed a semi-transparent window. (the image in the background is what picview should show, and is displayed in Teamcenter directly)
When opening the same picture directly from the same computer it works fine. Which makes no sense because Teamcenter just copies the file to the temp-folder and let’s windows open the picture.
So basically there is no difference.
I’d like to ask for an update that allows the user to disable the transparency of picview. Because I guess the transparency of the picview form might be the cause.

Thanks
Best regards,
Fred

From: Frederik Verscheure
Sent: dinsdag 25 juli 2023 11:50
To: Ruben2776/PicView ***@***.***>; Ruben2776/PicView ***@***.***>
Cc: Comment ***@***.***>
Subject: RE: [Ruben2776/PicView] Ask your questions here (Discussion #35)

Hi Ruben,

The problem is not the image file, since the picture opens on another computer in PicView without any problems.
Any idea what I can check? I use the portable version.

Grt
Fred

**Comment by thijsi123** [comment]
Is it by design that any changes are directly applied to the image like 'optimize image' etc...? I cancelled a crop saving and it immediately altered the image file itself.

**Reply by Ruben2776** [reply]
It is by design, yes. I personally don't like notifications and pop-ups very much, so it's designed to not be annoying, in my opinion.
I'm planning on making an undo feature, in a future update. E.g, if you overwrite an image, or delete it, I'm planning so that pressing Ctrl + Z would restore the original version.

I can't reproduce your crop cancelling issue, however the [recent preview version 7, and later,](https://github.com/Ruben2776/PicView/releases) as well as the upcoming version 3.0 have crop reworked. 

Also, the newer versions have an animated pop-up feature, that is currently used to warn you if you permanently delete an image and another pop-up, where you can choose if it warns you when closing the window by pressing Esc. 
I might extend its functionality to warnings, such as when cropping etc.

**Reply by thijsi123** [reply]
Cropping seems fixed for me, as for saving, I didn't know at first, so I was a little confused. Thanks for the reply.

**Comment by baiwuchang2022-SVIP** [comment]
1️⃣预选裁剪比例：自由 1:1 16:9 ...
2️⃣裁剪之后按快捷键Enter，自动进行下一张图像   ( 裁剪图像完成后自动保存指定的地址路径,默认格式为原始图像格式-无损裁剪。)
📝功能选项 "裁剪完成后自动移入回收站" "裁剪完成后自动彻底删除"是否使用。
📍原本我是不打算来评论的，但是在官网看到了"Lossless image optimization" Rotate, flip and crop. 感到震惊与欣慰，👍全网基本无一📍
💯希望可以采纳，之后请你喝杯☕

**Reply by hungrypancakemonster** [reply]
> 1️⃣预选裁剪比例：自由 1:1 16:9 ...

不好意思，这么说，你是想要有预选裁剪比例的意思吗？

> 2️⃣裁剪之后按快捷键Enter，自动进行下一张图像 ( 裁剪图像完成后自动保存指定的地址路径,默认格式为原始图像格式-无损裁剪。)

请问一下，你的意思是，想要裁剪过后，按Enter之后，就能够跳到下一张图（而裁剪过的图会自动保存）是吗？

> 📝功能选项 "裁剪完成后自动移入回收站" "裁剪完成后自动彻底删除"是否使用。

不好意思，你的意思是想要裁剪后有两个功能选项（以上所写的），是吗？你是想要裁剪过后删除那张图？

**Comment by Nymph666** [comment]
你好，我略缩图拉伸设置的均匀，但底部略缩图打开时会先显示成填充正方形，过一会才会变成均匀的形式，在软件中删除图片后会显示上一张图片，第一张图片删除后会直接跳转到最后一张，跳转到最后一张后略缩图是框选的倒数第二张，删除前面三张图片后开始，底部略缩图最后一张只保留空位，不显示了

https://github.com/user-attachments/assets/a05049d7-34d0-4b54-9aa0-0044c35a648f

**Comment by lllava** [comment]
Is it possible to move the window with the mouse by clicking anywhere on the picture? Now the window can be moved by clicking on the top bar.

**Reply by Ruben2776** [reply]
Yes, you can move it by holding down the `Shift` button and left mouse button
